render.c

```diff
@@ -656,18 +664,39 @@ render_clear_buffer(struct render_struct* p_render) {
 void
 render_double_up_lines(struct render_struct* p_render) {
   /* TODO: only need to double up partial lines within the render border. */
-  uint32_t line;
-
-  uint32_t lines = (p_render->height / 2);
   uint32_t width = p_render->width;
+  size_t line_size = width * sizeof(uint32_t);
   uint32_t double_width = (width * 2);
   uint32_t* p_buffer = p_render->p_buffer;
   uint32_t* p_buffer_next_line = (p_buffer + width);
-
-  for (line = 0; line < lines; ++line) {
-    (void) memcpy(p_buffer_next_line, p_buffer, (width * 4));
-    p_buffer += double_width;
-    p_buffer_next_line += double_width;
+  
+  if (p_render->is_double_size) {
+    int32_t line, column; /* must be signed */
```

Comment:
Style: one declaration per line. Comment should be init-capped, end with period.